pylightgbm/models.py

```diff
@@ -29,6 +29,7 @@ def __init__(self, exec_path="LighGBM/lightgbm", config="", application="regress
         self.config = config
         self.model = model
         self.verbose = verbose
+        self.num_iterations = num_iterations
```

Comment:
`self.num_iterations` is redundant with `self.param['num_iterations']`, try to use the latter.